internal/eval/eval.go

```diff
@@ -2759,6 +2759,36 @@ func evalHGETALL(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(results, false)
 }
 
+func evalHMGET(args []string, store *dstore.Store) []byte {
+	if len(args) < 2 {
+		return diceerrors.NewErrArity("HMGET")
+	}
+	key := args[0]
+
+	obj := store.Get(key)
+
+	var hashMap HashMap
+	results := make([]interface{}, len(args[1:]))
+
+	if obj != nil {
+		if err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeHashMap, object.ObjEncodingHashMap); err != nil {
+			return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)
+		}
+		hashMap = obj.Value.(HashMap)
+	}
+
```

Comment:
Hey, this might be considered as a nit, but I think we can return result as soon as the program knows obj is nil. What do you think about this suggestion?
```suggestion
	results := make([]interface{}, len(args[1:]))

	if obj == nil {
		return clientio.Encode(results, false)
	}

	if err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeHashMap, object.ObjEncodingHashMap); err != nil {
		return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)
	}
	var hashMap HashMap
	hashMap = obj.Value.(HashMap)

```